src/main/java/com/aws/greengrass/clientdevices/auth/connectivity/CISShadowMonitor.java

```diff
@@ -149,49 +208,38 @@ public void removeFromMonitor(CertificateGenerator certificateGenerator) {
-            }
-            try {
-                ShadowDeltaUpdatedSubscriptionRequest shadowDeltaUpdatedSubscriptionRequest =
-                        new ShadowDeltaUpdatedSubscriptionRequest();
-                shadowDeltaUpdatedSubscriptionRequest.thingName = shadowName;
-                iotShadowClient.SubscribeToShadowDeltaUpdatedEvents(shadowDeltaUpdatedSubscriptionRequest,
-                                QualityOfService.AT_LEAST_ONCE,
-                                this::processCISShadow,
-                                (e) -> LOGGER.atError()
-                                        .log("Error processing shadowDeltaUpdatedSubscription Response", e))
-                        .get(TIMEOUT_FOR_SUBSCRIBING_TO_TOPICS_SECONDS, TimeUnit.SECONDS);
-                LOGGER.info("Subscribed to shadow update delta topic");
-
-                GetShadowSubscriptionRequest getShadowSubscriptionRequest = new GetShadowSubscriptionRequest();
-                getShadowSubscriptionRequest.thingName = shadowName;
-                iotShadowClient.SubscribeToGetShadowAccepted(getShadowSubscriptionRequest,
-                                QualityOfService.AT_LEAST_ONCE, this::processCISShadow,
-                                (e) -> LOGGER.atError().log("Error processing getShadowSubscription Response", e))
-                        .get(TIMEOUT_FOR_SUBSCRIBING_TO_TOPICS_SECONDS, TimeUnit.SECONDS);
-                LOGGER.info("Subscribed to shadow get accepted topic");
-                return;
-
-            } catch (ExecutionException e) {
-                Throwable cause = e.getCause();
-                if (cause instanceof MqttException || cause instanceof TimeoutException) {
-                    LOGGER.atWarn().setCause(cause)
-                            .log("Caught exception while subscribing to shadow topics, will retry shortly");
-                } else if (cause instanceof InterruptedException) {
-                    throw (InterruptedException) cause;
-                } else {
-                    LOGGER.atError().setCause(e)
-                            .log("Caught exception while subscribing to shadow topics, will retry shortly");
-                }
-            } catch (TimeoutException e) {
-                LOGGER.atWarn().setCause(e).log("Subscribe to shadow topics timed out, will retry shortly");
-            }
-            // Wait for sometime and then try to subscribe again
-            Thread.sleep(WAIT_TIME_TO_SUBSCRIBE_AGAIN_IN_MS + JITTER.nextInt(10_000));
-        }
+    private void subscribeToShadowTopics() throws InterruptedException, ExecutionException {
+        ShadowDeltaUpdatedSubscriptionRequest shadowDeltaUpdatedSubscriptionRequest =
+                new ShadowDeltaUpdatedSubscriptionRequest();
+        shadowDeltaUpdatedSubscriptionRequest.thingName = shadowName;
+        iotShadowClient.SubscribeToShadowDeltaUpdatedEvents(
```

Comment:
let's do it in a different PR, but we should stop using the shadow client. It dynamically creates a new message handler every time it is called which will lead to duplicate subscription callbacks.

Reply:
ah geez, yep will follow up w/ a fix